packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1571,6 +1574,231 @@ abstract class InheritedWidget extends ProxyWidget {
   bool updateShouldNotify(covariant InheritedWidget oldWidget);
 }
 
+/// An [InheritedWidget] that's intended to be used as the base class for data
+/// models which are scoped to a widget tree.
+///
+/// An inherited widget's dependents are unconditionally rebuilt when the
+/// inherited widget changes per [InheritedWidget.updateShouldNotify].
+/// Widgets that depend on an [InheritedModel] qualify their dependence
+/// with an arbitrary "token" value. When the model is rebuilt, dependents
+/// will also be rebuilt, but only if there was a change in the model
+/// that corresponds to the token they provided.
+///
+/// An inherited model converts tokens into "dependencies" of type `T`
+/// with [getDependencyFor]. By default this method just returns its
+/// `token` argument. The inherited model's [updateShouldNotifyDependent]
+/// method is tested for each dependent and its dependencies.
+///
+/// Inherited widget descendants create a dependency on the inherited widget
+/// with [BuildContext.inheritFromWidgetOfExactType]. Typically this
+/// BuildContext method is called by a static `of` utility method
+/// defined by the [InheritedWidget] subclass. Inherited model dependencies
+/// are created similarly, with the addition of a parameter that's
+/// use as the dependency token. For example:
+///
+/// ```
+/// class MyModel extends InheritedModel<String> {
+///   // ...
+///   static MyModel of(BuildContext context, String dependency) {
+///     return context.inheritFromWidgetOfExactType(IntegerModel, token: dependency);
+///   }
+/// }
+/// ```
+///
+/// Calling `MyModel.of(context, 'foo')` means that `context` depends on the
+/// `foo` aspect of `MyModel`.
```

Comment:
Isn't it more accurate to say that it depends on MyModel, but only wants to be notified about changes relating to `foo`?
I could see calling the argument "aspect" or "subject" or some such. I'm not sure what `token` means in this context.